mlflow/models/model.py

```diff
@@ -22,12 +23,37 @@
     "Logging model metadata to the tracking server has failed, possibly due older "
     "server version. The model artifacts have been logged successfully under %s. "
     "In addition to exporting model artifacts, MLflow clients 1.7.0 and above "
-    "attempt to record model metadata to the  tracking store. If logging to a "
-    "mlflow server via REST, consider  upgrading the server version to MLflow "
+    "attempt to record model metadata to the tracking store. If logging to a "
+    "mlflow server via REST, consider upgrading the server version to MLflow "
     "1.7.0 or above."
 )
 
 
+class ModelInfo(NamedTuple):
+    run_id: str
+    artifact_path: str
+    model_uri: str
+    utc_time_created: str
+    flavors: Dict[str, Any]
+    model_uuid: str
+    saved_input_example_info: Dict[str, Any]
+    signature: ModelSignature
+
+
+ModelInfo.__doc__ = "The metadata of a logged MLflow Model."
+ModelInfo.run_id.__doc__ = "The ``run_id`` associated with the logged model."
+ModelInfo.artifact_path.__doc__ = "Run relative path identifying the logged model."
+ModelInfo.model_uri.__doc__ = "The ``model_uri`` of the logged model."
+ModelInfo.utc_time_created.__doc__ = "The UTC time that the logged model is created."
+ModelInfo.flavors.__doc__ = "Flavor module to save the model with."
+ModelInfo.model_uuid.__doc__ = "The ``model_uuid`` of the logged model."
+ModelInfo.saved_input_example_info.__doc__ = ""
+ModelInfo.signature.__doc__ = (
+    "A :py:class:`ModelSignature <mlflow.models.ModelSignature>` that "
```

Comment:
```suggestion
class ModelInfo(NamedTuple):
    """
    The metadata of a logged MLflow Model.
    """

    #: The metadata of a logged MLflow Model.
    run_id: str
    #: The metadata of a logged MLflow Model.
    artifact_path: str
    #: The metadata of a logged MLflow Model.
    model_uri: str
    #: The metadata of a logged MLflow Model.
    utc_time_created: str
    #: The metadata of a logged MLflow Model.
    flavors: Dict[str, Any]
    #: The metadata of a logged MLflow Model.
    model_uuid: str
    #: The metadata of a logged MLflow Model.
    saved_input_example_info: Dict[str, Any]
    #: A :py:class:`ModelSignature <mlflow.models.ModelSignature>` that
    #: describes the model input and output :py:class:`Schema <mlflow.types.Schema>`.
    signature: ModelSignature
```

Reply:
I think this is much cleaner than overwriting `__doc__`.

Reply:
Note `#:` is a special syntax for sphinx.

Reply:
And we should use `Optional` for properties that can be None (e.g. `signature`)

Reply:
Thanks! I updated the syntax.